alibi/utils/lang_model.py

```diff
@@ -0,0 +1,342 @@
+import abc
+import numpy as np
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+
+import tensorflow as tf  # type: ignore
+import transformers  # type: ignore
+from transformers import TFAutoModelForMaskedLM, AutoTokenizer
+
+
+class LanguageModel(abc.ABC):
+    SUBWORD_PREFIX = ''
+
+    def __init__(self, model_path: str, preloading: bool = True):
+        """
+        Initialize the language model.
+
+        Parameters
+        ----------
+        model_path
+            `transformers` package model path.
+        preloading
+            Whether to preload the online version of the transformer.
+            If `False`, a call to `load` method is expected.
+        """
+        self.model_path = model_path
+        self.model, self.caller, self.tokenizer = None, None, None
+
+        if preloading:
+            # set model
+            self.model = TFAutoModelForMaskedLM.from_pretrained(model_path)
+            self.caller = tf.function(self.model.call, experimental_relax_shapes=True)
+
+            # set tokenizer
+            self.tokenizer = AutoTokenizer.from_pretrained(model_path)
+
+    def from_disk(self, path: Union[str, Path]):
```

Comment:
How does this work? Shouldn't this be a class method that returns an initialized model?

Reply:
The `LanguageModel` has an `preloading` argument which is set by default to `True`. This download the online version and load it. If the `pretrained=False`, the model is not initialized and the call to `from_disk` is expected. It looks something like:

```python
model = BertBaseUncased(preloading=False)
model.from_disk(path)
```